### PR TITLE
Python 3 porting

### PR DESCRIPTION
**Pull request by benjwadams**
- print statements to functions, other changes to reflect Python 3 stdlib moves/renames/changes.
- Updated requirements in libraries for application compatibility with Python 3
- Convert unicode strings to bytes for BerkeleyDB password routines.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+FROM python:3.6-buster
+
+RUN apt-get update && apt-get -y install libxml2-dev libudunits2-dev
+RUN mkdir glider-dac && useradd glider
+COPY . /glider-dac
+WORKDIR glider-dac
+# not clear why reinstalling Mongo-related dependencies is necessary under
+# Python 3, but this allows the service to run without import or runtime errors
```
Comment (Bobfrat) [inline Dockerfile]: Does this happen with other base images or just python:3.6-buster?

File: glider_dac/models/deployment.py [hunk 2]
```diff
@@ -83,7 +83,7 @@ def dap(self):
             'user': slugify(self.username),
             'deployment': slugify(self.name)
         }
-        dap_url = u"http://%(host)s/thredds/dodsC/deployments/%(user)s/%(deployment)s/%(deployment)s.nc3.nc" % args
+        dap_url = "http://%(host)s/thredds/dodsC/deployments/%(user)s/%(deployment)s/%(deployment)s.nc3.nc" % args
```
Comment (Bobfrat) [inline glider_dac/models/deployment.py]: Do all these URLs (including the ones below) need to be https or is the ```host``` localhost?

Comment (benjwadams) [reply glider_dac/models/deployment.py]: `http://localhost` is still valid in the event `host` is `localhost`, so this still should work regardless.

Comment (Bobfrat) [reply glider_dac/models/deployment.py]: Sorry for confusion, the question was whether we should use https over http

**Pull request merged**